base/bucket.go

```diff
@@ -54,8 +54,25 @@ type BucketSpec struct {
 	Server, PoolName, BucketName, FeedType string
 	Auth                                   AuthHandler
 	CouchbaseDriver                        CouchbaseDriver
-	MaxNumRetries                          int // max number of retries before giving up
-	InitialRetrySleepTimeMS                int // the initial time to sleep in between retry attempts (in millisecond), which will double each retry
+	MaxNumRetries                          int     // max number of retries before giving up
+	InitialRetrySleepTimeMS                int     // the initial time to sleep in between retry attempts (in millisecond), which will double each retry
+	ViewQueryTimeoutSecs                   *uint32 // the view query timeout in milliseconds (default: 75000ms)
+}
+
+func (b BucketSpec) GetViewQueryTimeout() time.Duration {
+
+	// If the user doesn't specify any timeout, default to 75s
+	if b.ViewQueryTimeoutSecs == nil {
+		return time.Second * 75
+	}
+
+	// If the user specifies 0, then translate that to "No timeout"
+	if *b.ViewQueryTimeoutSecs == 0 {
+		return time.Hour * 24 * 7 * 365 * 10 // 10 years
```

Comment:
I think this is actually 70 years, but don't think it really matters in this context.